FirebaseAI/Tests/TestApp/Tests/Utilities/InstanceConfig.swift

```diff
@@ -19,62 +19,66 @@ import VertexAITestApp
+  static let vertexAI_v1 = InstanceConfig(
+    apiConfig: APIConfig(service: .vertexAI(endpoint: .firebaseProxyProd), version: .v1)
   )
-  static let vertexV1Staging = InstanceConfig(
-    apiConfig: APIConfig(service: .vertexAI(endpoint: .firebaseVertexAIStaging), version: .v1)
+  static let vertexAI_v1_staging = InstanceConfig(
+    apiConfig: APIConfig(service: .vertexAI(endpoint: .firebaseProxyStaging), version: .v1)
   )
-  static let vertexV1Beta = InstanceConfig(
-    apiConfig: APIConfig(service: .vertexAI(endpoint: .firebaseVertexAIProd), version: .v1beta)
+  static let vertexAI_v1beta = InstanceConfig(
+    apiConfig: APIConfig(service: .vertexAI(endpoint: .firebaseProxyProd), version: .v1beta)
   )
-  static let vertexV1BetaStaging = InstanceConfig(
-    apiConfig: APIConfig(service: .vertexAI(endpoint: .firebaseVertexAIStaging), version: .v1beta)
+  static let vertexAI_v1beta_staging = InstanceConfig(
+    apiConfig: APIConfig(service: .vertexAI(endpoint: .firebaseProxyStaging), version: .v1beta)
   )
-  static let developerV1Beta = InstanceConfig(
-    apiConfig: APIConfig(service: .developer(endpoint: .firebaseVertexAIProd), version: .v1beta)
+  static let googleAI_v1beta = InstanceConfig(
+    apiConfig: APIConfig(service: .googleAI(endpoint: .firebaseProxyProd), version: .v1beta)
   )
-  static let developerV1BetaStaging = InstanceConfig(
-    apiConfig: APIConfig(service: .developer(endpoint: .firebaseVertexAIStaging), version: .v1beta)
+  static let googleAI_v1beta_staging = InstanceConfig(
+    apiConfig: APIConfig(service: .googleAI(endpoint: .firebaseProxyStaging), version: .v1beta)
   )
-  static let developerV1Spark = InstanceConfig(
+  static let googleAI_v1_freeTier_bypassProxy = InstanceConfig(
     appName: FirebaseAppNames.spark,
-    apiConfig: APIConfig(service: .developer(endpoint: .generativeLanguage), version: .v1)
+    apiConfig: APIConfig(service: .googleAI(endpoint: .googleAIBypassProxy), version: .v1)
   )
-  static let developerV1BetaSpark = InstanceConfig(
+  static let googleAI_v1beta_freeTier_bypassProxy = InstanceConfig(
     appName: FirebaseAppNames.spark,
-    apiConfig: APIConfig(service: .developer(endpoint: .generativeLanguage), version: .v1beta)
+    apiConfig: APIConfig(service: .googleAI(endpoint: .googleAIBypassProxy), version: .v1beta)
   )
-  static let allConfigs = [
-    vertexV1,
-    vertexV1Staging,
-    vertexV1Beta,
-    vertexV1BetaStaging,
-    developerV1Beta,
-    developerV1BetaStaging,
-    developerV1Spark,
-    developerV1BetaSpark,
-  ]
 
-  static let allConfigsExceptDeveloperV1 = [
-    vertexV1,
-    vertexV1Staging,
-    vertexV1Beta,
-    vertexV1BetaStaging,
-    developerV1Beta,
-    developerV1BetaStaging,
-    developerV1BetaSpark,
+  static let allConfigs = [
+    vertexAI_v1,
+    vertexAI_v1_staging,
+    vertexAI_v1beta,
+    vertexAI_v1beta_staging,
+    googleAI_v1beta,
+    googleAI_v1beta_staging,
+    googleAI_v1_freeTier_bypassProxy,
```

Comment:
Isn't freeTier separate from bypassing the proxy? We could have free tier with the proxy too

Reply:
@paulb777 Good point. We couldn't when I set these up initially but we can now. I'll merge as-is but tomorrow I'll onboard our free tier project for Firebase AI. Let's also discuss offline whether we even want to continue integration testing in the bypass proxy case. It's not a scenario we'll be shipping, though possibly it could tell us if there's ever a Google AI problem vs. a proxy problem. The free tier quota will be quite limited, which is why I'm reluctant to run all tests both ways.